File: Swift/powerful-integers.swift
class Solution {
    func powerfulIntegers(_ x: Int, _ y: Int, _ bound: Int) -> [Int] {
        var res = Set<Int>()
        let logX = x != 1 ? Int((log(Double(bound)) / log(Double(x)))) + 1 : 1
        let logY = y != 1 ? Int((log(Double(bound)) / log(Double(y)))) + 1 : 1
        var powX = 1

        for _ in 0..<logX {
            var powY = 1
            for _ in 0..<logY {
                let val = powX + powY
                if val <= bound {
                    res.insert(val)
                }
                powY *= y
            }
            powX *= x
        }

        return Array(res)
    }
}

File: Python/powerful-integers.py
import math

class Solution:
    def powerfulIntegers(self, x, y, bound):
        """
        :type x: int
        :type y: int
        :type bound: int
        :rtype: List[int]
        """
        res = set()
        logX = int(math.floor(math.log(bound) / math.log(x))) + 1 if x != 1 else 1
        logY = int(math.floor(math.log(bound) / math.log(y))) + 1 if y != 1 else 1
        powX = 1

        for _ in range(logX):
            powY = 1
            for _ in range(logY):
                val = powX + powY
                if val <= bound:
                    res.add(val)
                powY *= y
            powX *= x

        return list(res)
